core/lib/spree/money.rb

```diff
@@ -92,6 +94,22 @@ def as_json(*)
       to_s
     end
 
+    def <=>(other)
+      if !other.respond_to?(:money)
+        raise TypeError, "Can't compare #{other.class} to Spree::Money"
+      end
+      if self.currency != other.currency
+        # By default, ::Money will try to run a conversion on `other.money` and
+        # try a comparison on that. We do not want any currency conversion to
+        # take place so we'll catch this here and raise an error.
+        raise(
+          DifferentCurrencyError,
+          "Can't compare #{self.currency} with #{other.currency}"
+        )
+      end
+      self.money <=> other.money
```

Comment:
`@money` would work as this is a `attr_reader`, won't work for `@currency`, though, as this is a delegate to `money`

👍 for being consistent on `@money` and `currency` usage. 

🤔 thinking to enable Rubocops "redundant self" checks...